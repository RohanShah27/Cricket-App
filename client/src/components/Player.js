import React, { Component } from 'react'
import { connect } from "react-redux";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";
// import DatePicker from 'react-date-picker/dist/entry.nostyle';
import { createPlayers } from "../actions/Players";
import { getPlayertype } from "../actions/playerType";
import "../styles/player.css"
class Player extends Component {
    constructor() {
        super();
        this.handleDayChange = this.handleDayChange.bind(this);
        this.state = {
            gender: "male",
            player_role: "choice"

        };
        this.handleChange = this.handleChange.bind(this);
        this.onChange = this.onChange.bind(this);
        this.onCreate = this.onCreate.bind(this);
        this.handleTypeChange = this.handleTypeChange.bind(this);
    }
    handleChange(e) {
        this.setState({ gender: e.target.value });
    }

    handleTypeChange(e) {
        this.setState({ player_role: e.target.value });
    };
    componentDidMount() {
        this.props.getPlayertype();
        if (!localStorage.getItem("token")) {
            this.props.history.push("/");
        }
    }
    state = {
        player_name: "",
        gender: "",
        player_dob: new Date(),
        player_nation: "",
        player_role: "",
        batting_style: "",
        bowling_style: "",
    }
    onChange = event => {
        this.setState({ [event.target.name]: event.target.value });
    };
    onCreate() {
        let player = {
            player_name: this.state.player_name,
            gender: this.state.gender,
            player_dob: this.state.player_dob.toLocaleDateString('en-GB'),
            batting_style: this.state.batting_style,
            bowling_style: this.state.bowling_style,
            player_role: this.state.player_role,
            player_nation: this.state.player_nation
        };


        this.props.createPlayers(player);
        this.setState({
            player_name: "",
            gender: "",
            player_dob: "",
            player_nation: "",
            player_role: "",
            batting_style: "",
            bowling_style: "",
        });
    }

    handleDayChange = date => {
        this.setState({ player_dob: date });
    };
    render() {
        return (

            < div >
                {/* <h1>{this.props.players_type.player_type}</h1> */}
                {/* <p>{this.props.players_type.map(player => (<p>{player.player_type}</p>))}</p> */}
                <form id="playerform">
                    <fieldset>

                        <div className="PlayerAdding">
                            <h1 className="playerheader">Add Player</h1>
                            <input className="playerdetails"
                                type="text"
                                name="player_name"
                                placeholder=" Player Name"
                                onChange={this.onChange}
                                value={this.state.player_name}
                            />

                            <div className="playerflex-container">
                                <div className="division">
                                    <select value={this.state.gender} onChange={this.handleChange} className="genders">
                                        <option name="choice" className="choice">Gender</option>
                                        <option name="male"> Male</option>
                                        <option name="female">Female</option>
                                    </select>
                                </div>
                                <div className="division">
                                    <DatePicker className="playerdate"
                                        dateFormat="dd-MM-yyyy"
                                        placeholderText="DOB"
                                        selected={this.state.player_dob}
                                        onChange={this.handleDayChange}
                                        popperPlacement="bottom"
                                        popperModifiers={{
                                            flip: {
                                                behavior: ["bottom"] // don't allow it to flip to be above
                                            },
                                            preventOverflow: {
                                                enabled: false // tell it not to try to stay within the view (this prevents the popper from covering the element you clicked)
                                            },
                                            hide: {
                                                enabled: false // turn off since needs preventOverflow to be enabled
                                            }
                                        }}
                                    />
                                </div>
                            </div>
                            <p>
                                <input className="playerdetails"
                                    type="text"
                                    name="player_nation"
                                    placeholder="Nationality"
                                    onChange={this.onChange}
                                    value={this.state.player_nation}
                                />{" "}
                            </p>
                            <p>
                                <select value={this.state.player_role} onChange={this.handleTypeChange} className="playerType">
                                    <option name="choice">player type</option>
                                    {this.props.playerType.map(playerType => (

                                        <option name="Batsman">{playerType.player_role}</option>

                                    ))}

                                </select>
                            </p>
                            <p>
                                <input className="playerdetails"
                                    type="text"
                                    name="batting_style"
                                    placeholder="Batting Style"
                                    onChange={this.onChange}
                                    value={this.state.batting_style}
                                />
                            </p>
                            <p>
                                <input className="playerdetails"
                                    type="text"
                                    name="bowling_style"
                                    placeholder="Bowling Style"
                                    onChange={this.onChange}
                                    value={this.state.bowling_style}
                                />
                            </p>
                            {/* <p>
                <input className="playerdetails"
                  type="text"
                  name="playerdescription"
                  placeholder=" Description"
                  onChange={this.onChange}
                  value={this.state.playerdescription}
                />
              </p> */}
                        </div>
                        {this.props.error ? <><p>{this.props.error}</p></> : null}
                        <button onChange={this.onChange} onClick={this.onCreate} className="playerbutton">
                            Add Player
              </button>
                    </fieldset>
                </form >
            </div >
        )
    }
}
const mapStateToProps = state => ({
    Adminplayers: state.AdminplayerReducer.Adminplayers,
    playerType: state.playertypeReducer.playerType,
    error: state.userReducer.error

});

export default connect(
    mapStateToProps,
    { createPlayers, getPlayertype }
)(Player);

