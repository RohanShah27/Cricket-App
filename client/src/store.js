import { createStore, applyMiddleware, compose } from "redux";

import thunk from "redux-thunk";
import rootReducer from "./reducers";

const initialState = {};
//think will give all redducer actions to the middleware object
const middleware = [thunk];
// will give access to the extension
//   //We do this to check if chrome has the redux extension or not if not then it installs the devtoolextension in chrome
const composeEnhancers =
  typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
      })
    : compose;
const enhancer = composeEnhancers(applyMiddleware(...middleware));
const store = createStore(rootReducer, initialState, enhancer);
export default store;
