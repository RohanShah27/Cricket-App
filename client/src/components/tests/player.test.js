import React from "react";
import { mount } from "enzyme";
import Player from "../player";
import { Provider } from "react-redux";
import store from "../../store";

const player = jest.fn();
const wrapper = mount(
    <Provider store={store}>
        <Player player={player} history={[]} />
    </Provider>
);

describe("Test player Component", () => {
    it("should render the component", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should have exactly 5 input fields", () => {
        expect(wrapper.find("input").length).toBe(5);
    });

    it("should have exactly one button", () => {
        expect(wrapper.find("button").length).toBe(1);
    });

    it("should have exactly Add Player text on Button", () => {
        expect(wrapper.find("button").text()).toEqual("Add Player");
    });

    it("should have exactly two select option", () => {
        expect(wrapper.find("select").length).toBe(2);
    });


});
