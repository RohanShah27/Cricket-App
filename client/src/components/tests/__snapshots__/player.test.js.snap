// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test player Component should render the component 1`] = `ReactWrapper {}`;
